include/dx2/reflection.hpp

```diff
@@ -0,0 +1,373 @@
+#pragma once
+
+#include "dx2/h5/h5read_processed.hpp"
+#include "dx2/h5/h5write.hpp"
+#include <experimental/mdspan>
+#include <functional>
+#include <memory>
+#include <optional>
+#include <string>
+#include <vector>
+
+#pragma region Utils
+/*
+ * Type aliases for centralised control of storage types.
+ */
+
+// Internal storage type for the data
+template <typename T> using storage_type = std::vector<T>;
+// Type alias for mdspan (currently using backport std::experimental::mdspan)
+template <typename T>
+using mdspan_type =
+    std::experimental::mdspan<T, std::experimental::dextents<size_t, 2>>;
+
+namespace reflection_table_type_utils {
+template <typename T> struct type_tag_t {
+  using type = T;
+};
+template <typename T> constexpr type_tag_t<T> type_tag{};
+} // namespace reflection_table_type_utils
+
+/**
+ * @brief Dispatches based on the type of an HDF5 dataset.
+ *
+ * This function inspects the type of the dataset identified by
+ * `dataset_id`, and then calls the provided callback as a templated
+ * function with the appropriate C++ type.
+ *
+ * @tparam Callback A callable with a templated `operator()<T>()`.
+ * @param dataset_id An open HDF5 dataset identifier (hid_t).
+ * @param cb The callback to invoke with the deduced template type.
+ */
+template <typename Callback>
+void dispatch_h5_dataset_type(hid_t dataset_id, Callback &&cb) {
+  using namespace reflection_table_type_utils;
+
+  // Get the dataset's datatype
+  hid_t type_id = H5Dget_type(dataset_id);
+  // Get the class of the datatype (e.g., integer, float)
+  H5T_class_t cls = H5Tget_class(type_id);
+  // Get the size of the datatype
+  size_t size = H5Tget_size(type_id);
+
+  // Match to supported types
```

Comment:
(This is looking really nice btw!) I was giving this branch a test on a dials strong.refl (h5 format), and we also need to be able to handle arrays which are 64bit unsigned int (e.g. flags, panel). I am currently reading these like
`std::vector<std::size_t> flags = read_array_from_h5_file<std::size_t>(.....)`.

Reply:
Thanks for adding handling for these extra type!